Library/Homebrew/cmd/readall.rb

```diff
@@ -34,14 +38,23 @@ def readall_args
   def readall
     args = readall_args.parse
 
+    if args.no_simulate?
+      # odeprecated "--no-simulate", "nothing (i.e. not passing `--os` or `--arch`)"
+    end
+
     if args.syntax? && args.no_named?
       scan_files = "#{HOMEBREW_LIBRARY_PATH}/**/*.rb"
       ruby_files = Dir.glob(scan_files).grep_v(%r{/(vendor)/})
 
       Homebrew.failed = true unless Readall.valid_ruby_syntax?(ruby_files)
     end
 
-    options = { aliases: args.aliases?, no_simulate: args.no_simulate? }
+    options = {
+      aliases:     args.aliases?,
+      no_simulate: args.no_simulate?,
+    }
+    # TODO: Always pass this once `--os` and `--arch` are passed explicitly to `brew readall` in CI.
```

Comment:
This needs to wait for CI to get updated in other repos, right? Looks like we'll be able to get rid of the options hash as well which is nice.